### PR TITLE
Rework Markdown parsing

### LINKED ISSUE #193: Markdown newlines are different from Element
The original Markdown spec describes that newlines should be removed from the input, unless 2 spaces are appended at the end (line break), or more than 1 newlines (new paragraph).
However, many applications use a different flavor, e.g. Github, or Element, where every newline is persisted as is.
This is useful in instant messaging.
Cinny should also preserve these line breaks.
image
image

### LINKED ISSUE #709: Do not treat everything between two $ as TeX math
Typing `$1 instead of $2` should not be treated as TeX math

> The opening $ must
have a non-space character immediately to its right, while the closing $ must
have a non-space character immediately to its left, and must not be followed
immediately by a digit. Thus, $20,000 and $30,000 won’t parse as math.

### LINKED ISSUE #749: Numbered lists are numbered wrongly
### Describe the bug

sending
``` 
5. five
10. ten
```
gives
image

### Reproduction

```
"content": {
    "body": "5. five\n10. ten",
    "format": "org.matrix.custom.html",
    "formatted_body": "<ol start=\"5\">\n<li>five</li>\n<li>ten</li>\n</ol>\n",
    "msgtype": "m.text",
    "org.matrix.msc1767.message": [
      {
        "body": "5. five\n10. ten",
        "mimetype": "text/plain"
      },
      {
        "body": "<ol start=\"5\">\n<li>five</li>\n<li>ten</li>\n</ol>\n",
        "mimetype": "text/html"
      }
    ]
  }
```

### Expected behavior

Element does this
image
which in my opinion is (at least/only) half correct by following the suggested `start` property of `ol` https://spec.matrix.org/latest/client-server-api/#mroommessage-msgtypes

### Platform and versions

```shell
OS: Arch
Browser: 5.4.2763.3 (Official Build) snapshot (64-bit) 
Cinnyv2.1.2
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ginnyTheCat**
### Description
Simplify markdown parsing to allow for new features and better spec compliance.

- [x] Spoilers
  - [x] `||spoiler||` syntax
  - [x] new `||spoiler||(reason)` syntax
    - [ ] #370
  - [ ] plain text fallback
- [x] Inline math using `$1+1`
  - [x] respecting rules outlined in #709
- [x] Display math using `$$1+1$$`
  - [x] without newlines `$$1+1$$` (new)
  - [x] with newlines `$$\n1+1\n$$`
- [ ] Plain text rendering

Fixes #709
Fixes #193
Fixes #749 (replicate element behavior)

#### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

Preview: https://6302170673d245070a1fa41c--pr-cinny.netlify.app
⚠️ Exercise caution. Use test accounts. ⚠️

### CODE REVIEW

**Comment by ajbura** [comment]
Is it ready for review?

**Comment by ginnyTheCat** [comment]
i think it's ready.
the plaintext rendering is not completed yet as making `||test||` being sent as `[spoiler](mxc://spoiler/content-here)` would make it annoying and confusing to edit messages with a spoiler as this is what would show up in the edit field. I'm thinking about how to mitigate that but atm the only possibility i see is parsing the html version.
but we can merge the pr and add that later.

**Comment by ginnyTheCat** [comment]
Sorry for messing up the commits a bit. Accidentally did a rebase which clashed with your merge. Hope it's ok now

File: src/app/molecules/message/Message.scss [hunk 1]
```diff
@@ -168,7 +173,11 @@
   & > .text > a {
     white-space: initial !important;
   }
-  
+
+  & p:not(:last-child) {
+    margin-bottom: var(--sp-normal);
+  }
+
```
Comment (ajbura) [inline src/app/molecules/message/Message.scss:180]: Can we replace \n with `<br />` instead of applying client specific css for newlines?

Comment (ginnyTheCat) [reply src/app/molecules/message/Message.scss:180]: this is to space `<p>` tags further apart to make them appear as actual paragraphs (as shown in the screenshots of #193). this does not relate to `\n` or `<br>` afaik.

**Review by ajbura — APPROVED** [review]
Looks great. Thank you :).

**Pull request merged**